crates/subspace-core-primitives/src/lib.rs

```diff
@@ -244,3 +251,39 @@ impl RootBlock {
         }
     }
 }
+
+/// Encoded block with mapping of objects that it contains
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub struct EncodedBlockWithObjectMapping {
+    /// Encoded block
+    #[cfg_attr(feature = "std", serde(with = "HexForm"))]
+    pub block: Vec<u8>,
+    /// Mapping of objects inside of the block
+    pub object_mapping: BlockObjectMapping,
+}
+
+/// Metadata necessary for farmer operation
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub struct FarmerMetadata {
```

Comment:
I prefer it `FarmerConfig` or something as metadata is sort of a special word for me.
```suggestion
pub struct FarmerConfig {
```

Reply:
Arguably this is closer to the core primitives, but not really. Another data structure that kind of belongs to a separate crate altogether.

It is called metadata because it is not a static configuration, but rather something that is derived from the blockchain itself. Maybe metadata is not the best word, but I don't feel like this is a configuration either.

Reply:
How about `FarmerProperty`? Also not good at naming though :D ...